src/sorting/binary_insertion_sort.rs

```diff
@@ -0,0 +1,57 @@
+fn binary_search<T: Ord>(arr: &[T], target: &T) -> usize {
+    let mut low = 0;
+    let mut high = arr.len();
+
+    while low < high {
+        let mid = low + (high - low) / 2;
+
+        if arr[mid] < *target {
+            low = mid + 1;
+        } else {
+            high = mid;
+        }
+    }
+
+    low
+}
+
+fn binary_insertion_sort<T: Ord + Clone>(arr: &mut [T]) {
+    let len = arr.len();
+
+    for i in 1..len {
+        let key = arr[i].clone();
+        let index = binary_search(&arr[..i], &key);
+        
+        arr[index..i+1].rotate_right(1);
+        arr[index] = key;
+    }
+}
+
+fn main() {
```

Comment:
Please remove `main`, there are tests to test the implementation

Reply:
Fixed it. I apologize for forgetting to remove the main function. I will be sure to remove main function in my future contributions

Reply:
No problem